packages/backend-next/src/plugins/search.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendModule,
+} from '@backstage/backend-plugin-api';
+import { searchIndexRegistryExtensionPoint } from '@backstage/plugin-search-backend-node/alpha';
+import { DefaultCatalogCollatorFactory } from '@backstage/plugin-catalog-backend';
+import { DefaultTechDocsCollatorFactory } from '@backstage/plugin-techdocs-backend';
+import { ToolDocumentCollatorFactory } from '@backstage/plugin-explore-backend';
+import { loggerToWinstonLogger } from '@backstage/backend-common';
+
+export const searchIndexRegistry = createBackendModule({
+  moduleId: 'searchIndexRegistry',
+  pluginId: 'search',
+  register(env) {
+    env.registerInit({
+      deps: {
+        indexRegistry: searchIndexRegistryExtensionPoint,
+        config: coreServices.config,
+        discovery: coreServices.discovery,
+        tokenManager: coreServices.tokenManager,
+        logger: coreServices.logger,
+        scheduler: coreServices.scheduler,
+      },
+      async init({
+        indexRegistry,
+        config,
+        logger,
+        discovery,
+        tokenManager,
+        scheduler,
+      }) {
+        const schedule = scheduler.createScheduledTaskRunner({
+          frequency: { minutes: 10 },
+          timeout: { minutes: 15 },
+          // A 3 second delay gives the backend server a chance to initialize before
+          // any collators are executed, which may attempt requests against the API.
+          initialDelay: { seconds: 3 },
+        });
+
+        indexRegistry.addCollator({
+          schedule,
+          factory: DefaultCatalogCollatorFactory.fromConfig(config, {
+            discovery,
+            tokenManager,
+          }),
+        });
```

Comment:
Is it possible to move each of these into separate modules that can be installed? They shouldn't really be provided by the respective plugins btw, but perhaps that could be a separate fix.

Reply:
😮 , I love that idea, lets see how much time we have to be able to do this as part of the migration! But otherwise definitely shortly after!

Reply:
me and @camilaibs will work on getting that in tomorrow as part of this PR as well!

Reply:
we have now added this as part of the last commit https://github.com/backstage/backstage/pull/16710/commits/4c8abe964f67e5a7db5b390a7a162e98aee4472a have a look when you have time @Rugvip

Reply:
Done!